src/content/en/showcase/2017/santa.md

```diff
@@ -0,0 +1,217 @@
+
+In 2016, we upgraded the engine behind Santa Tracker to support an offline experience for most scenes.
+This excludes scenes backed by YouTube videos or those that deal with Santa's live location—which is of course only available with a direct connection to the North Pole! 📶☃️
+
+<figure>
+  <img src="/web/showcase/2017/images/santa/phone.png" />
+  <figcaption>
+    Santa Tracker on an Android device
+  </figcaption>
+</figure>
+
+## Challenges
+
+Santa incorporates a responsive design that works well across phones, tablets and desktop.
+The site delights with great multimedia, including stylized visuals and holiday-themed audio.
+However, a regular build of Santa Tracker is several hundred mb (!).
+This is for a few reasons-
+
+* Santa is supported in over 35 languages, so many assets must be duplicated
+* Different platforms have different media support (e.g., mp3 vs ogg)
+* Multimedia files are sometimes provided in different sizes and resolutions.
+
+Santa's elves are also hard at work throughout December, often releasing new, critical updates throughout the month.
+This means that assets already cached by a user's browser may need to be refreshed ton repeat visits.
+
+These challenges;
+
+* Large multimedia resources for different 'scenes'
+* Changes released throughout the month
+
+...result in the unsuitability of a naïve offline strategy.
+
+## Santa, built with Polymer
+
+It's worth stepping back and talking about Santa's overall design before diving into how we upgraded it to an offline PWA.
+
+Santa is a single-page application, originally written in Polymer 0.5, and now upgraded to Polymer 1.7.
+Santa is made up of a shared set of code—the router, shared nav assets etc—but it also has many unique 'scenes'.
+
+<figure class="attempt-right">
+  <img src="/web/showcase/2017/images/santa/preloader.gif"  />
+</figure>
+
+Each scene is accessible through a different URL—`/village.html`, `/codelab.html` and `/boatload.html`—and is its own web component.
+When a user opens a scene, we preload all its required HTML and assets (images, audio, css, js), which exist under `/scenes/[[sceneName]]` in the Santa Tracker repository.
+While that happens, users see a friendly preloader that shows progress.
+
+This approach means that we don't have to load unnecessary assets for scenes the user doesn't see (which is a lot of data!).
+It also means that we need to keep an internal 'cache manifest' (keep this in mind—we'll refer to it later) of all the assets required for every scene.
+
+### Load What You Use
+
+This model saves bandwidth, only serving resources required for the scenes a user visits: rather than preloading the whole site at once.
+Santa Tracker leverages Polymer's ability to upgrade custom elements at runtime, rather than at load time-
+
+```html
+<lazy-pages id="lazypages" selected-item="{{selectedScene}}" ... >
```

Comment:
DevSite gets cranky with `{{` template tags, you need to encode one of them, see [here](https://web-central.appspot.com/web/resources/style-guide#special_case_templates_-)